packages/workflow-core/src/lib/plugins/common-plugin/risk-rules-plugin.ts

```diff
@@ -0,0 +1,89 @@
+import groupBy from 'lodash.groupby';
+import { TContext } from '../../utils';
+import { RiskRulesPluginParams } from './types';
+import { logger } from '../../logger';
+
+export class RiskRulePlugin {
+  public static pluginType = 'risk-rules';
+  name: RiskRulesPluginParams['name'];
+  rulesSource: RiskRulesPluginParams['rulesSource'];
+  stateNames: RiskRulesPluginParams['stateNames'];
+  action: RiskRulesPluginParams['action'];
+  successAction: RiskRulesPluginParams['successAction'];
+  errorAction: RiskRulesPluginParams['errorAction'];
+
+  constructor(pluginParams: RiskRulesPluginParams) {
+    this.name = pluginParams.name;
+    this.stateNames = pluginParams.stateNames;
+    this.rulesSource = pluginParams.rulesSource;
+    this.action = pluginParams.action;
+    this.successAction = pluginParams.successAction;
+    this.errorAction = pluginParams.errorAction;
+  }
+
+  async invoke(context: TContext) {
+    try {
+      const rulesetResult = await this.action(context, this.rulesSource);
```

Comment:
Learning from past experience, I think we should add logging to the `try` and not just the `catch` so we'd have better understanding what happens in the plugin

Reply:
totally agree!,
we also should inject sentry and notify us for critical issues 
already seen when notion api wasnt respond